Source/Engine/Physics/PhysicsScene.h

```diff
@@ -0,0 +1,560 @@
+#pragma once
+
+#include "SimulationEventCallback.h"
+
+#include "Engine/Scripting/ScriptingObject.h"
+#include "Engine/Scripting/ScriptingType.h"
+#include "Engine/Core/Collections/Array.h"
+#include "Engine/Core/Math/Vector3.h"
+#include "Engine/Core/Math/Quaternion.h"
+#include "Types.h"
+#include <ThirdParty/PhysX/PxPhysicsAPI.h>
```

Comment:
I'd prefer to not include any PhysX headers in `PhysicsScene.h`

Reply:
OK, I got rid of the include by implementing a pimpl instead. C++ isn't my strongest language so if you have a preferred alternate method of resolving your request let me know and I'll implement that instead.